src/note.rs

```diff
@@ -209,18 +209,26 @@ impl Note {
         sponge_hash(&[sk_r, pos])
     }
 
-    /// Return a hash represented by `H(value_commitment, pos, H([R]),
-    /// H([pskr]))`
+    /// Return a hash represented by `H(note_type, value_commitment,
+    /// H(StealthAddress), pos, encrypted_data)
     pub fn hash(&self) -> BlsScalar {
         let value_commitment = self.value_commitment().to_hash_inputs();
         let pk_r = self.stealth_address().pk_r().to_hash_inputs();
+        let R = self.stealth_address().R().to_hash_inputs();
 
         sponge_hash(&[
+            BlsScalar::from(self.note_type as u64),
             value_commitment[0],
             value_commitment[1],
-            BlsScalar::from(self.pos()),
+            BlsScalar::from(self.nonce),
             pk_r[0],
             pk_r[1],
+            R[0],
+            R[1],
+            BlsScalar::from(self.pos()),
+            self.encrypted_data.cipher()[0],
```

Comment:
Add the cipher as separate variable, something like:
```rust
         let value_commitment = self.value_commitment().to_hash_inputs();
         let pk_r = self.stealth_address().pk_r().to_hash_inputs();
         let R = self.stealth_address().R().to_hash_inputs();
         let cipher = self.encrypted_data.cipher();
```

and then:

            R[1],
            BlsScalar::from(self.pos()),
            cipher[0],
            cipher[1],
            cipher[2]
```

Also, add a comment where here we're assuming `cipher` has length 3. We could add an assert but it would be an additional check currently we don't need, and I've in mind an alternative solution.